framework/src/main/java/org/checkerframework/common/accumulation/AccumulationTransfer.java

```diff
@@ -0,0 +1,151 @@
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+/**
+ * The default transfer function for an accumulation checker.
+ *
+ * <p>Subclasses should call the {@link #accumulate(Node, TransferResult, String...)} accumulate} or
+ * {@link #accumulate(MethodInvocationNode, TransferResult, String...)} methods to accumulate a
+ * string at a particular program point.
+ */
+public class AccumulationTransfer extends CFTransfer {
+
+    protected final AccumulationAnnotatedTypeFactory typeFactory;
+
+    public AccumulationTransfer(CFAnalysis analysis) {
+        super(analysis);
+        typeFactory = (AccumulationAnnotatedTypeFactory) analysis.getTypeFactory();
+    }
+
+    /**
+     * Updates the estimate of how many things node has accumulated to include all the values.
+     *
+     * @param node the node whose estimate should be adjusted
+     * @param result the result containing the store to be modified
+     * @param values the new accumulation values
+     */
+    public void accumulate(Node node, TransferResult<CFValue, CFStore> result, String... values) {
+        AnnotatedTypeMirror oldType = typeFactory.getAnnotatedType(node.getTree());
+        AnnotationMirror newAnno = getNewAnno(oldType, values);
+        insertIntoStores(result, node, newAnno);
+    }
+
+    /**
+     * Updates the estimate for the receiver and any other receiver-returning methods in a chain
+     * with this MethodInvocationNode to include the new values.
+     *
+     * <p>For example, if the argument is the expression {@code a.b().c()}, the new value is "foo",
+     * and b and c return their receiver, all of the expressions {@code a}, {@code a.b()}, and
+     * {@code a.b().c()} would have their estimates updated to include "foo".
+     *
+     * @param node a method invocation whose receiver is to be updated
+     * @param result the result containing the store to be modified
+     * @param values the new accumulation values
+     */
+    public void accumulate(
+            MethodInvocationNode node, TransferResult<CFValue, CFStore> result, String... values) {
+        Node receiver = node.getTarget().getReceiver();
+        AnnotatedTypeMirror oldType = typeFactory.getReceiverType(node.getTree());
+        // e.g. if the node being visited is static
+        if (oldType == null) {
+            return;
+        }
+        AnnotationMirror newAnno = getNewAnno(oldType, values);
+        while (receiver != null) {
+            insertIntoStores(result, receiver, newAnno);
```

Comment:
As noted above, this call will do nothing when `receiver` is a method call that is not deterministic.  Not sure if we should change the implementation logic to reflect this more clearly.  Nothing incorrect happens now, but it may confuse future readers of the code